src/src/beneficiaries/beneficiaries.service.ts

```diff
@@ -297,10 +305,276 @@ if(!result){
+      if (coreBeneficiarieskeyExist.length > 0) {
+        response[i++] = await this.hasuraService.q(
+          'core_beneficiaries',
+          {
+            ...req,
+            id: req?.core_beneficiaries?.id
+              ? req?.core_beneficiaries?.id
+              : null,
+            user_id: user_id,
+          },
+          coreBeneficiariesArr,
+          update,
+        );
+      }
+    }
+    return this.userById(user_id);
+  }
+
+  async newCreate(req: any) {
+    const tableName = 'users';
+    const newR = await this.hasuraService.q(tableName, req, [
+      'first_name',
+      'last_name',
+      'mobile',
+      'lat',
+      'long',
+      'keycloak_id',
+    ]);
+    const user_id = newR[tableName]?.id;
+    if (user_id) {
+      await this.hasuraService.q(`beneficiaries`, { ...req, user_id }, [
+        'facilitator_id',
+        'user_id',
+      ]);
+      await this.hasuraService.q(`core_beneficiaries`, { ...req, user_id }, [
+        'device_ownership',
+        'device_type',
+        'user_id',
+      ]);
+    }
+    return await this.userById(user_id);
+  }
+
+  async userById(id: any) {
+    var data = {
+      query: `query searchById {        
+        users_by_pk(id: ${id}) {
+          id
+          first_name
+          last_name              
+          dob
+          mobile
+          grampanchayat
+          village
+          block
+          district
+          state
+          state_id
+          aadhar_no
+          aadhar_token
+          aadhar_verified
+          address
+          alternative_mobile_number
+          block
+          profile_url
+          block_id
+          district_id
+          email_id
+          gender
+          lat
+          long
+          block_village_id
+          beneficiaries {
+            beneficiaries_found_at
+            created_by
+            facilitator_id
+            id
+            program_id
+            rsos_id
+            updated_by
+          }
+          core_beneficiaries {
+            career_aspiration
+            updated_by
+            type_of_learner
+            status
+            reason_of_leaving_education
+            previous_school_type
+            mobile_ownership
+            learner_wish_to_pursue_education
+            last_standard_of_education_year
+            last_standard_of_education
+            last_school_type
+            id
+            connect_via_refrence
+            created_by
+            device_ownership
+            device_type
+            document_id
+            enrolled_for_board
+            enrollement_status
+          }
+          extended_users {
+            marital_status
+            designation
+            created_by
+            id
+            user_id
+            updated_by
+            social_category
+            qualification_id
+          }
+        }}`,
+    };
+
+    const response = await lastValueFrom(
```

Comment:
import this service from service/hasura.service.ts. don't repeat the same code everywhere. Please reuse it.